Summarization.py

```diff
@@ -11,155 +11,168 @@
-file = open(r'C:/Users/wb550776/Downloads/Using ML in Evaluative Synthesis May 31 2019.docx')
-#file = open(file , 'r')
-text = file.read()
-tokenized_sentence = sent_tokenize(text)
-sentences = tokenized_sentence
-
-text = remove_special_characters(str(text))
-text = re.sub(r'\d+', '', text)
-tokenized_words_with_stopwords = word_tokenize(text)
-tokenized_words = [word for word in tokenized_words_with_stopwords if word not in Stopwords]
-tokenized_words = [word for word in tokenized_words if len(word) > 1]
-tokenized_words = [word.lower() for word in tokenized_words]
-
-# output to excel
-writer = pd.ExcelWriter(os.path.join('tables', 'document_prediction.xlsx'))
-df.to_excel(writer, sheet_name='project_prediction')
-result_list[0].to_excel(writer, sheet_name='category1_salient_prediction')
-# df_result_topic.to_excel(writer, sheet_name='salient_prediction_by_topic')
-writer.save()
+    for s in text:
+      sentences.append(sent_tokenize(s))
+    
+    sentences = [y for x in sentences for y in x] # flatten list
+    
+    #for s in df['article_text']:
+    #  sentences.append(sent_tokenize(s))
+    
+    #!wget http://nlp.stanford.edu/data/glove.6B.zip
+    #!unzip glove*.zip
+    
+    # Extract the word vectors
+    #cd C:\Users\wb550776\Documents\Projects\Summarization
+    word_embeddings = {}
+    
+    f = open(r'C:\Users\wb550776\Documents\IEG_Sumerization\glove.6B.100d.txt', encoding='utf-8')
+    for line in f:
+        values = line.split()
+        word = values[0]
+        coefs = np.asarray(values[1:], dtype='float32')
+        word_embeddings[word] = coefs
+    f.close()
+    
+    len(word_embeddings)
+    #We now have word vectors for 400,000 different terms stored in the dictionary – ‘word_embeddings’.
+    
+    #################
+    #Text Preprocessing
+    # remove punctuations, numbers and special characters
+    clean_sentences = pd.Series(sentences).str.replace("[^a-zA-Z]", " ")
+    
+    # make alphabets lowercase
+    clean_sentences = [s.lower() for s in clean_sentences]
+    
+    #Get rid of Stopwords
+    nltk.download('stopwords')
+    from nltk.corpus import stopwords
+    stop_words = stopwords.words('english')
+    
+    #define a function to remove the stopwords from the dataset:
+    # function to remove stopwords
+    def remove_stopwords(sen):
+        sen_new = " ".join([i for i in sen if i not in stop_words])
+        return sen_new
+    
+    
+    # remove stopwords from the sentences
+    clean_sentences = [remove_stopwords(r.split()) for r in clean_sentences]
+    
+    
+    ####################################
+    ##Vector Representation of Sentences
+    
+    # Extract word vectors
+    word_embeddings = {}
+    f = open(r'C:\Users\wb550776\Documents\IEG_Sumerization\glove.6B.100d.txt', encoding='utf-8')
+    for line in f:
+        values = line.split()
+        word = values[0]
+        coefs = np.asarray(values[1:], dtype='float32')
+        word_embeddings[word] = coefs
+    f.close()

```

Comment:
It looks like a duplication of code from above. You probably want to abstract this away into a top-level function `get_word_embeddings()`